toolboxes/mri_core/mri_core_utility.cpp

```diff
@@ -6,8 +6,246 @@
 
 #include "mri_core_utility.h"
 #include "hoNDArray_elemwise.h"
+#include "mri_core_kspace_filter.h"
+#include <ctime>
 
 namespace Gadgetron
 {
+    template <typename T>
+    void detect_readout_sampling_status(const hoNDArray<T>& data, hoNDArray<float>& sampled)
```

Comment:
Not sure I get these comments. sampled is an hoNDArray? And I don't think this function should have an array as return value.